target/linux/x86/image/Makefile

```diff
@@ -43,6 +44,9 @@ endif
 SIGNATURE:=$(shell perl -e 'printf("%08x", rand(0xFFFFFFFF))')
 ROOTPART:=$(call qstrip,$(CONFIG_TARGET_ROOTFS_PARTNAME))
 ROOTPART:=$(if $(ROOTPART),$(ROOTPART),PARTUUID=$(SIGNATURE)-02)
+EFI_SIGNATURE:=$(strip $(shell uuidgen | sed -r 's/[a-zA-Z0-9]{2}$$/00/'))
```

Comment:
Like `SIGNATURE` above it, this will impact build reproduceability; is that okay?

Reply:
@DHowett What are the other choices?  Something unique but deterministic, like the SHA of HEAD?

Reply:
Thinking about this more -- if there's already nondeterminism in the build system, perhaps this isn't the time for me to suggest avoiding adding more. I remember that this was discussed on the mailing list at one point. Since these IDs are used for unequivocally identifying disks, reusing them in the name of determinism might be "dangerous" (for some definition of the word dangerous).

If there are other options in the config tree for fixing identifiers (stripping path names, setting timestamps, etc.) the partition signatures could be one more option, but that might just be something to be done in the future.

Feel free to resolve this comment as WONTFIX. :smile:

Reply:
@DHowett Use deterministic UUIDs, RFC 4122 Section 4.3, quoting:

> The requirements for these types of UUIDs are as follows:
>  *  The UUIDs generated at different times from the same name in the same namespace MUST be equal.